provider/pkg/await/await.go

```diff
@@ -90,17 +90,19 @@ type ReadConfig struct {
 
 type UpdateConfig struct {
 	ProviderConfig
-	Previous *unstructured.Unstructured
-	Inputs   *unstructured.Unstructured
-	Timeout  float64
-	Preview  bool
+	OldInputs  *unstructured.Unstructured
+	OldOutputs *unstructured.Unstructured
+	Inputs     *unstructured.Unstructured
```

Comment:
I'd suggest splitting the await logic change into a separate PR since it's not immediately apparent from the changelog or commit message that it's changing in a fundamental way.

Alternatively, you could structure this as two commits / changelog entries in the same PR, and then not squash merge them.

Reply:
This seems like a good proposal. While the actual logic to enable `generateName` is dependent on the await logic changes, the reverse isn't true. I'm more in favour of splitting these into 2 separate PRs, with the await logic being merged first.